ide/svg/src/org/netbeans/modules/svg/navigation/SVGNavigatorPanel.java

```diff
@@ -0,0 +1,208 @@
+     * current context to work on
+     */
+    private Lookup.Result<SVGDataObject> currentContext;
+    /**
+     * listener to context changes
+     */
+    private LookupListener contextListener;
+    /**
+     * Listens for changes on image file.
+     */
+    private FileChangeListener fileChangeListener;
+    private long lastSaveTime = -1;
+    private DataObject currentDataObject;
+    private static final RequestProcessor WORKER = new RequestProcessor(SVGNavigatorPanel.class.getName());
+
+    @Override
+    public String getDisplayName() {
+        return NbBundle.getMessage(SVGNavigatorPanel.class, "Navigator_DisplayName");
+    }
+
+    @Override
+    public String getDisplayHint() {
+        return NbBundle.getMessage(SVGNavigatorPanel.class, "Navigator_DisplayHint");
+    }
+
+    @Override
+    public JComponent getComponent() {
+        if (lastSaveTime == -1) {
+            lastSaveTime = System.currentTimeMillis();
+        }
+        if (panelUI == null) {
+            panelUI = new SVGPreviewPanel();
+        }
+        return panelUI;
+    }
+
+    @Override
+    public void panelActivated(Lookup context) {
+        // lookup context and listen to result to get notified about context changes
+        currentContext = context.lookup(MY_DATA);
+        currentContext.addLookupListener(getContextListener());
+        // get actual data and recompute content
+        Collection<? extends SVGDataObject> data = currentContext.allInstances();
+        currentDataObject = getDataObject(data);
+
+        if (currentDataObject == null) {
+            return;
+        }
+
+        if (fileChangeListener == null) {
+            fileChangeListener = new SVGFileChangeAdapter();
+        }
+        currentDataObject.getPrimaryFile().addFileChangeListener(fileChangeListener);
+        setNewContent(currentDataObject);
+    }
+
+    @Override
+    public void panelDeactivated() {
+        currentContext.removeLookupListener(getContextListener());
+        currentContext = null;
+        if (currentDataObject != null) {
+            currentDataObject.getPrimaryFile().removeFileChangeListener(fileChangeListener);
+        }
+        currentDataObject = null;
+    }
+
+    @Override
+    public Lookup getLookup() {
+        // go with default activated Node strategy
+        return null;
+    }
+
+    private void setNewContent(final DataObject dataObject) {
+        if (dataObject == null) {
+            return;
+        }
+
+        WORKER.post(() -> {
+            FileObject fo = dataObject.getPrimaryFile();
+
+            if (fo == null) {
+                return;
+            }
+
+            if (panelUI == null) {
+                getComponent();
+            }
+
+            SVGLoader svgLoader = new SVGLoader();
+            SVGDocument svgDocument = svgLoader.load(fo.toURL());
+
+            if (svgDocument == null) {
+                return;
+            }
+
+            SwingUtilities.invokeLater(() -> panelUI.setSVG(svgDocument));
```

Comment:
navigator needs to update even when svgDocument is null, otherwise it will keep displaying the last preview even if the selected file is broken - this could be confusing. The null check can be simply removed i think.